cli/jobs/pipelines-with-components/image_classification_with_densenet/image_cnn_train/main.py

```diff
@@ -582,7 +583,11 @@ def _worker_init_fn(id):
     # override the master node ip by intention
     # args.dist_url = 'tcp://' + get_master_ip() + ':23456'
     # extract master ip from os env as a workaround
-    args.dist_url = "tcp://" + os.environ["AZ_BATCHAI_MPI_MASTER_NODE"] + ":23456"
+    if "AZUREML_CR_DISTRIBUTED_CONFIG" in os.environ:
```

Comment:
Looking at the component for this Pipeline Step MPI based multi-node launching is being used, though it doesn't seem like the training actually directly leverages MPI, it just uses the rank variables to set world info (e.g. OMPI_COMM_WORLD_SIZE).
With CR this example would be better off using the pytorch distribution configuration, it can also be configured to launch 1 process per node like MPI is, and it sets these env vars:
MASTER_ADDR
MASTER_PORT
WORLD_SIZE
RANK
LOCAL_RANK
NODE_RANK
It doesn't require MPI to be installed, which is also a bonus.

Reply:
Updated the yaml file to use pytorch to launch the job.